docs/helm-broker/02-02-deep-dive.md

```diff
@@ -0,0 +1,41 @@
+---
+title: Architecture deep dive
+type: Architecture
+---
+
+The diagram and steps describe the Helm Broker workflow in details, including the logic of its inner components, namely the Controller and Broker.
+
+![Architecture deep dive](./assets/hb-deep-dive.svg)
+
+1. The Controller watches for ClusterAddonsConfiguration (CAC) and AddonsConfiguration (AC) custom resources (CRs).
+2. The user creates, updates, or deletes CAC or AC custom resources.
+3. The Controller fetches and parses the data of all addon repositories defined under the **spec.repositories** field in a given CR. During this step, the Controller does the following:
+  - Analyze fetched addons against errors.
```

Comment:
```suggestion
  - Analyzes fetched addons against errors.
```
Analyzes against errors? Don't you mean that it "checks the fetched addons for errors"?